index.js

```diff
@@ -1,7 +1,9 @@
-export default async function pDoWhilst(action, condition) {
-	const actionResult = await action();
+export default async function pDoWhilst(action, condition, initialValue) {
+	const actionResult = await action(await initialValue);
```

Comment:
`initialValue` should not await the promise. That makes it impossible for the `initialValue` to be a promise you actually want passed into `action`, if you want to deal with the promise there.

Reply:
You are right! If `initialValue` rejects for some reason, the exception will be thrown inside `pDoWhilst` preventing `action` do anything about it.